zkp/circuits/lib/check-hashes-value.circom

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2024 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+pragma circom 2.1.4;
+
+include "../node_modules/circomlib/circuits/poseidon.circom";
+include "../node_modules/circomlib/circuits/comparators.circom";
+
+// CheckHashesValue is a circuit that checks the integrity of transactions of Fungible Tokens
+//   - check that all output values are positive numbers (within the range of 0 to 2^40)
+//   - check that the output commitments are the hash of the output values
+//   - check that the sum of output values equals a total value in the output
```

Comment:
?? we just add them up and return the output from what I can see

Reply:
yes this is a confusing aspect of ZKP circuits:
- the proof generator calculates the output, which is included in the proof object
- the output is always part of the public input parameters used to verify the proof
- the verifier independently assembles the public input parameters, in this case the expected value from the `amount` parameter of the transaction function
- if the proof verification succeeds, then the verifier knows that the values represented by the output UTXOs are exactly equal to the `amount`

Reply:
Awesome - happy about that